application/comit_node/src/bin/comit_node.rs

```diff
@@ -56,7 +57,14 @@ fn main() -> Result<(), failure::Error> {
 }
 
 fn load_settings() -> Result<ComitNodeSettings, config::ConfigError> {
-    let comit_config_path = var_or_default("COMIT_NODE_CONFIG_PATH", "~/.config/comit_node".into());
+    let user_dirs = directories::UserDirs::new().unwrap();
```

Comment:
🔧 I'd prefer a `Err` being returned than panicking. Try to replace `config::ConfigError` in the returned `Result` with `failure::Error`.

Reply:
I have rewritten the function to avoid the panic! problem.
While browsing through the code, i have seen a few other places where the tilde is used to point to the home path!